tests/integration/test_load_and_run_multi_agent_a2a_tool.py

```diff
@@ -0,0 +1,206 @@
+            return str(datetime.datetime.now())
+
+        date_agent_description = "Agent that can return the current date."
+        date_agent_cfg = AgentConfig(
+            instructions="Use the available tools to obtain additional information to answer the query.",
+            name="date_agent",
+            model_id=agent_model,
+            description=date_agent_description,
+            tools=[get_datetime],
+            model_args=model_args,
+        )
+        date_agent = await AnyAgent.create_async(
+            agent_framework=agent_framework,
+            agent_config=date_agent_cfg,
+            tracing=TracingConfig(console=False, cost_info=True),
+        )
+
+        # SERVING PROPER
+
+        served_agent = date_agent
+        (served_task, served_server) = await served_agent.serve_async(
+            serving_config=ServingConfig(
+                port=tool_agent_port,
+                endpoint=f"/{tool_agent_endpoint}",
+                log_level="info",
+            )
+        )
+
+        # Search agent is ready for card resolution
+
+        logger.info(
+            "Setting up agent",
+            extra={
+                "endpoint": f"http://localhost:{tool_agent_port}/{tool_agent_endpoint}"
+            },
+        )
+
+        main_agent_cfg = AgentConfig(
+            instructions="Use the available tools to obtain additional information to answer the query.",
+            description="The orchestrator that can use other agents via tools using the A2A protocol.",
+            tools=[
+                await a2a_query(
+                    f"http://localhost:{tool_agent_port}/{tool_agent_endpoint}"
+                )
+            ],
+            model_args=model_args,
+            **kwargs,  # type: ignore[arg-type]
+        )
+
+        main_agent = await AnyAgent.create_async(
+            agent_framework=agent_framework,
+            agent_config=main_agent_cfg,
+            tracing=TracingConfig(console=False, cost_info=True),
+        )
+
+        agent_trace = await main_agent.run_async(
+            # "Which LLM agent framework is the most appropriate to execute SQL queries using grammar constrained decoding? I am working on a business environment on my own premises, and I would prefer hosting an open source model myself."
+            "What date and time is it right now?"
+        )
+
+        await asyncio.sleep(5)
+
+        assert isinstance(agent_trace, AgentTrace)
+        assert agent_trace.final_output
+
+        logger.info("spans:")
+        logger.info(agent_trace.spans)
+
+        logger.info(agent_trace.final_output)
+        now = datetime.datetime.now()
+        assert all(
+            [
+                str(now.year) in agent_trace.final_output,
+                str(now.day) in agent_trace.final_output,
+                now.strftime("%B") in agent_trace.final_output,
+            ]
+        )
+        assert any(
+            span.attributes["gen_ai.operation.name"] == "execute_tool"
```

Comment:
I think we should use these helpers so if/when we need to update the logic to check whether a span is a tool the test will still work.

```suggestion
            span.is_tool_execution()
```